tests/post_training/pipelines/lm_weight_compression.py

```diff
@@ -157,6 +161,25 @@ def prepare_model(self) -> None:
                     self.fp32_model_dir, trust_remote_code=True, load_in_8bit=False, compile=False, stateful=is_stateful
                 )
             self.model = self.model_hf.model
+        elif self.backend == BackendType.ONNX:
+            if is_stateful:
+                msg = f"is_stateful={is_stateful} is not supported for {self.backend.value} backend."
+                raise RuntimeError(msg)
+
+            if not (self.fp32_model_dir / self.ONNX_MODEL_NAME).exists():
+                opset_version = self.params.get("opset", None)
+                if opset_version:
+                    main_export(self.model_id, self.fp32_model_dir, opset=opset_version)
+                    self.model_hf = ORTModelForCausalLM.from_pretrained(self.fp32_model_dir, trust_remote_code=True)
+                else:
+                    self.model_hf = ORTModelForCausalLM.from_pretrained(self.model_id, export=True)
+                    self.model_hf.save_pretrained(self.fp32_model_dir)
```

Comment:
We can't specify the opset version with `ORTModelForCausalLM.from_pretrained()`. To obtain a model with an opset version lower than 21, we should use the `main_export()` method (or a similar approach).